x/gamm/twap/module.go

```diff
@@ -0,0 +1,137 @@
+package twap
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gorilla/mux"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/keeper"
+	twaptypes "github.com/osmosis-labs/osmosis/v7/x/gamm/twap/types"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
+
+type AppModuleBasic struct {
+	cdc codec.Codec
+}
+
+func (AppModuleBasic) Name() string { return twaptypes.ModuleName }
+
+func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+}
+
+func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	return json.RawMessage{}
+	// return cdc.MustMarshalJSON(types.DefaultGenesis())
+}
+
+// ValidateGenesis performs genesis state validation for the gamm module.
+func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
+	// var genState types.GenesisState
+	// if err := cdc.UnmarshalJSON(bz, &genState); err != nil {
+	// 	return fmt.Errorf("failed to unmarshal %s genesis state: %w", types.ModuleName, err)
+	// }
+	// return genState.Validate()
+	return nil
+}
+
+//---------------------------------------
+// Interfaces.
+func (b AppModuleBasic) RegisterRESTRoutes(ctx client.Context, r *mux.Router) {
+}
+func (b AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+	types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx)) //nolint:errcheck
+}
+
+func (b AppModuleBasic) GetTxCmd() *cobra.Command {
+	return nil
+	// return cli.NewTxCmd()
+}
+func (b AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return nil
+	// return cli.GetQueryCmd()
+}
+
+// RegisterInterfaces registers interfaces and implementations of the gamm module.
+func (AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+}
+
+type AppModule struct {
+	AppModuleBasic
+
+	ak types.AccountKeeper
+	bk types.BankKeeper
+	gk keeper.Keeper
+	tk twapkeeper
```

Comment:
Consider expanding abbreviations for readability

Reply:
This is standard within the SDK